db/resync_status.go

```diff
@@ -0,0 +1,105 @@
+package db
+
+import (
+	"sync"
+)
+
+type ResyncManager struct {
+	Status     ResyncStatus
+	LastError  error
+	Terminator bool       // Allows resync operation to be cancelled while in progress
+	Mutex      sync.Mutex // Used to lock the Status, LastError and Terminator
+}
+
+type ResyncStatus struct {
+	Status        string `json:"status"`
+	DocsProcessed int    `json:"docs_processed"`
+	DocsChanged   int    `json:"docs_changed"`
+	Error         string `json:"last_error,omitempty"`
+}
+
+const (
+	ResyncStateRunning  = "running"
+	ResyncStateStopped  = "stopped"
+	ResyncStateStopping = "stopping"
+	ResyncStateError    = "stopped on error"
```

Comment:
What do you think about simplifying this to just 'error'?  Better to be consistent in how we report error status (consistent with replicationStatus in this case).

Reply:
I can make that change if you prefer but I think "stopped on error" is much less ambiguous for some extra text. If its just "error" this could mean it hit an error and will retry or it hit an error but will carry on.